### ISSUE TITLE
RTL8187 Driver Older Kernels

### ISSUE THREAD

**Issue by creited** [meefik/linuxdeploy#473]
Hello,

after great acknowledge using Kali Distro, i decided to give it a try on my old tablet (Brazilian brand) which has built in 4.1 Jelly Bean and 3.0.8 Kernel.

After some research I was able to have Kali running on my tablet (SANA) and it works like a charm. But, I am not able to compile drivers for the USB Dongle RTL8187. I understand that these drivers are built in for most of the kernels but for my specific one these drivers are not available. 

Since the tablet is quite old, I am not able to update for a newer kernel from the manufacturer.

When I try the method found on google, and try to make the drivers I get this error:

`/lib/modules/3.0.8/build: No such file or directory.  Stop.`

Checking the folders, there is no [3.0.8] folder inside modules.

Googleing again, I tried and was able to download linux headers and image for kernels 4~ . (From Kali's source list)

Is it possible to compile the drivers using these images/headers? How should I proceed?

Is is possible to download image/headers for my current 3.0.8 version? I searched and I couldn't find any.

What do you specialists suggest me?

Thank you guys!

**Comment by sedit10n** [comment]
Your device manufacturer is supposed to provide kernel sources on their
device.  The Google kernel was forked from the  original kernel by Linus
that you can get from kernel.org.  check your device manufacturer Web site
for the kernel sources. Which will include all necessary patches and
configuration file specifically for yours. Then you should be able to build
modules for it.  You may find it on github as well.  So yes you can build
custom modules.  Your device may have prebuilt modules in your initramfs
or /system partion.

adb shell to your device and find / -name "*.ko" then you can copy them to
the kali mount.

**Comment by creited** [comment]
Thanks for the reply!

I will be happy to try https://www.kernel.org/pub/linux/kernel/v3.0/ this weekend.

Since I'm a begginer, any advices on how should I proceed with one of these files?

**Comment by creited** [comment]
So, I was able to find some "*.ko" inside /system/modules/3.0.8-default/ accessing my tablet with adb.
I couldn't find any drivers for RTL8187 there. 

What should I do with these ko files? Do I need to send them to kali environment that im running?
The only one way I was able to copy files, was: copying them to my C:/folder/ in windows, then I zipped the folder and sent by mail. After putting all those KO files inside /lib/modules/3.0.8/ and running lsmod, those new files didn't update on those modules showed.

**Comment by creited** [comment]
Still no success. Any help?

**Comment by creited** [comment]
Have any one walked int this trouble? Still no success...

**Comment by ACIDIXx** [comment]
See if your usb wifi board is discover, with this command : 
`lsusb` 
try to install debian's drivers : 
`apt-get install wireless-tools`
and verify if you have a wireless extension device who is detected by your linux kernel with : 
`iwconfig`
to identify an usb device, follow this link https://wiki.debian.org/fr/HowToIdentifyADevice/USB (Frensh)

**Comment by creited** [comment]
Should I update it on Linux Deploy environment or android environment? Any suggestions on this?

I really appreciate your help on it. Now I will start my research on "How to update kernel" instead of "How to compile drivers".

Have a nice day!

**Comment by ACIDIXx** [comment]
`uname -r` and post the output, just to be on the safe side

**Comment by creited** [comment]
This tells me my kernei ls:  `3.0.8` and just it.

Another information I was able to gather using Android Device Analyzer:

wmt.model.no
SungWorld_MID7_W730_GSL1680_Z7Z67-030A

My running Android Version is 4.1.1 JB.

Im not sure, but it seems to be running WonderMedia ROM. (WM8850 or something, im not sure how to check it).

**Comment by ACIDIXx** [comment]
http://android.stackexchange.com/questions/47472/how-can-i-update-the-linux-kernel-of-my-android-device
> Normally, your kernel would be updated when the operating system - your "ROM" - is updated

Try to update your ROM to install android KitKat 4.4
Tuto : http://kikatmodupdatenow.blogspot.fr/2015/07/working-tutorial-how-to-updateupgrade_17.html

**Comment by creited** [comment]
I was definetely beaten by this filesquick.net file hoster. I wasn`t abble to complete any survey, or bypass it (also tryed the video posted on the tutorial/VPN, etc). Is it possible to download any other ROM from any other place? what should it be specific for?

Thanks

**Comment by creited** [comment]
I don't think this file would work, since I couldn`t find any other tutorial to update wm8850 to kitkat and when I tried to look for Lillipop for WM8850 I found this site http://blgnewdlnow.blogspot.com.br/2015/07/working-tutorial-how-to-updateupgrade_45.html which takes you to the same survey system to download..... 

Would you know any other method to upload my kernel/android version so that I can use RTL drivers?

Thanks
